keps/sig-cluster-lifecycle/generic/1755-communicating-a-local-registry/README.md

```diff
@@ -0,0 +1,593 @@
+  
+- A general-purpose mechanism for publicly exposing cluster configuration.
+
+- A general-purpose mechanism for cluster capability detection for developer
+  tooling. With the best practices around cluster capability detection moving
+  fast, local registries are the only best practice that's mature enough where
+  this makes sense.
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation.  The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+Tools that configure a local registry should also apply a ConfigMap that
+communicates "LocalRegistryStatus".
+
+The ConfigMap specifies everything a tool might need to know about how to
+interact with the local registry.
+
+Any tool that pushes an image to a registry should be able to read this
+ConfigMap, and decide to push to the local registry instead.
+
+### User Stories (optional)
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system.  The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story 1
+
+Alice is setting up a development environment on Kubernetes.
+
+Alice creates a local Kind cluster. She follows the Kind-specific instructions
+for setting up a local registry during cluster creation.
+
+Alice will have some tool she interacts with that builds images and pushes them
+to a registry the cluster can access. That tool might be an IDE like VSCode,
+an infra-as-code toolchain like Pulumi, or a multi-service dev environment like Tilt.
+
+On startup, the tool should connect to the cluster and read the `local-registry`
+ConfigMap. The ConfigMap embeds a versioned config.
+
+If the config specifies a registry location, the tool may automatically adjust
+image pushes to push to the specified registry.
+
+If the config specifies a `help` URL, the tool may prompt the user to set up a
+registry for faster development.
+
+### Notes/Constraints/Caveats (optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above.
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+- Users may see the ConfigMap and draw mistaken conclusions about how to
+  interact with it. They think deleting the ConfigMap would delete the local
+  registry (which it does not).
+
+- This KEP does not specify any automatic mechanism for keeping the
+  LocalRegistryStatus up-to-date with how the cluster is configured.
+  For example, the user might delete the registry. The cluster might
+  not have a way of knowing that the registry has died.
+  
+- This KEP creates the new API version `local-registry.x-k8s.io/v1alpha1`,
```

Comment:
this is out of date WRT the new ConfigMap format, potentially remove this point?

Reply:
removed!